pkg/util/k8s/tags.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"regexp"
+
+	"antrea.io/nephe/pkg/cloudprovider/securitygroup"
+	"antrea.io/nephe/pkg/converter/target"
+	"antrea.io/nephe/pkg/labels"
+)
+
+const TagCharacterClass = `^[a-zA-Z0-9][\.\-a-zA-Z0-9_]*[a-zA-Z0-9]$`
+
+// ImportTags function returns tags which comply with kubernetes labels format.
+func ImportTags(tags map[string]*string) map[string]string {
+	importedTags := make(map[string]string)
+	// Tags are used as kubernetes labels in External Entity in nephe.io/tag-<tag key>=<tag value> format.
+	// Filter out the tags which don't comply with kubernetes label format.
+	// For Label format, refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set.
+	for key, val := range tags {
+		value := *val
+		if len(key) > (target.LabelSizeLimit - len(labels.ExternalEntityLabelKeyTagPrefix)) {
+			continue
+		}
+		regExp, err := regexp.Compile(TagCharacterClass)
+		if err != nil {
+			continue
+		}
+		if matched := regExp.MatchString(key); !matched {
+			continue
+		}
+
+		if len(value) > target.LabelSizeLimit {
+			continue
+		}
+		if len(value) > 0 {
+			if matched := regExp.MatchString(value); !matched {
+				continue
+			}
+		}
+
+		// skip any tags added by nephe for internal processing
+		_, hasAGPrefix, hasATPrefix := securitygroup.IsNepheControllerCreatedSG(key)
```

Comment:
We should be having this kind of check there.. If you want to skip some tags, pass it as function argument.

Reply:
added in caller.